### PR TITLE
docs: :building_construction: diagrams of internals for `path_sprout_root()`

### LINKED ISSUE #657: Create internal flow diagram for `path_sprout_root()`

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

These changes add a diagram for the `path_sprout_root()`.

Closes #657

## Reviewer Focus

This PR needs an in-depth review.

### CODE REVIEW

**Review by martonvago — APPROVED** [review]
Makes sense 👍 👍 !

File: docs/design/implementation/images/core/path-sprout-root.puml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+@startuml path-sprout-root
+!theme seedcase from https://raw.githubusercontent.com/seedcase-project/seedcase-theme/main
+
+state "path_sprout_root()" as prs {
+
+  state "get_root_envvar()" as gre
+  gre : - If user sets the `SPROUT_ROOT`\n  environment variable
```
Comment (signekb) [inline docs/design/implementation/images/core/path-sprout-root.puml:7]: How will the user set this variable? :)

Comment (lwjohnst86) [reply docs/design/implementation/images/core/path-sprout-root.puml:7]: It's described in the Guide doc right now ☺️

Comment (signekb) [reply docs/design/implementation/images/core/path-sprout-root.puml:7]: Ah, yeah! That's right. Sorry, I forgot!

**Review by signekb — APPROVED** [review]

**Pull request merged**